src/GreenPipes/IContextSource.cs

```diff
@@ -0,0 +1,10 @@
+﻿using System.Threading;
+using System.Threading.Tasks;
+
+namespace GreenPipes
+{
+    public interface IContextSource<TContext> where TContext : class, PipeContext
+    {
+        Task Fill(IPipe<TContext> pipe, CancellationToken token = default(CancellationToken));

```

Comment:
Well...




    // similar to the IBusObserver
    public interface ISourceObserver<T>
    {
        Task PostCreate(ISourceContext...);

        Task PreStart(ISource...);
        Task PostStart(...);
        Task StartFault(..., Exception exception);

        Task PreStop();
        Task PostStop();
        Task StopFault(..., Exception exception);
    }